### PR TITLE
update loader to support google maps 3d tiles

### PR DESCRIPTION
**Pull request by kfarr**
Work in progress, remaining items before merging:

Pre-merge list:
- [x] @Algorush modify readme docs to add info about new component properties
- [x] for 3DStreet use case, research how to implement 'reduce draw distance' option -- currently way too many tiles are loaded in the distance that are not visible in this use case (workaround by setting `distanceScale` to 0.01, see comment below)
- [x] @Avnerus to review
- [x] consider adding warning and instructions to docs regarding google policy for default 3d tiles root json load quota https://github.com/nytimes/aframe-loader-3dtiles-component/issues/40
- [x] @Algorush test removing `preserveDrawingBuffer`

Not resolving for this PR (future):
* for 3DStreet use case, research how to switch to inspector camera and not pause the component when in inspector / editor mode. Not part of this ticket, track here (https://github.com/nytimes/aframe-loader-3dtiles-component/issues/16#issuecomment-1136382446))
* Would it make sense to include [depandabot version bumps](https://github.com/nytimes/aframe-loader-3dtiles-component/pulls?q=is%3Aopen+is%3Apr+dependabot+) from here? (No, not part of this ticket)
* add this to the /examples/ index file with a screen shot? later after this https://github.com/nytimes/aframe-loader-3dtiles-component/issues/42
* test in VR mode and re-add teleport controls? https://github.com/nytimes/aframe-loader-3dtiles-component/issues/44

### CODE REVIEW

**Comment by kfarr** [comment]
We had an issue where way too many tiles are loaded in the distance than are needed for this application. There are not visible as they are mostly occluded by buildings in the scene given the camera placement. For users, the issue can be both the draw area is too big and/or the amount of bandwidth transfer for these tiles is excessive for the use case. 

Instead of a radius-based drawing limit, if we just focus on MB of data transferred the `distanceScale` option addresses this. Testing different values for `distanceScale` over the same use case of ~45 seconds of usage of the application resulted in the following:
* 0.01 = 10MB for 45-60 seconds of heavy usage;
* 0.1 = 20MB;
* 1.0 = 60MB

Therefore setting distanceScale: 0.01 (for our use case) will fix the throughput issue in the short-term and later we can research drawing only in a specified area.

File: examples/google-tiles/index.html [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    <a-scene vr-mode-ui="enabled: false" renderer="colorManagement: true;">
+      <a-entity camera="fov:45; near:1; far: 1000" look-controls="reverseMouseDrag: true" wasd-controls="enabled: true" id="camera" cursor-teleport="cameraRig: #camera; cameraHead: #camera;"></a-entity>
+      <a-entity
+        id="tileset"
+        loader-3dtiles="
+          url: https://tile.googleapis.com/v1/3dtiles/root.json; 
+          lat: 37.77522354250163;
+          long: -122.41931773049723;
+          height: -16.5;
+          googleApiKey: INSERTYOURKEYHERE;
```
Comment (Avnerus) [inline examples/google-tiles/index.html:41]: The example demo states that this and also lat/long/height are available parameters but there is no option to insert them via the query string.

File: index.js [hunk 2]
```diff
@@ -28,24 +28,33 @@ AFRAME.registerComponent('loader-3dtiles', {
...
     this.camera = this.data.cameraEl?.object3D.children[0] ?? document.querySelector('a-scene').camera;
     if (!this.camera) {
       throw new Error('3D Tiles: Please add an active camera or specify the target camera via the cameraEl property');
     }
     const { model, runtime } = await this._initTileset();
+
     this.el.setObject3D('tileset', model);
 
     this.originalCamera = this.camera;
+    this.el.sceneEl.renderer.preserveDrawingBuffer = true;
```
Comment (Avnerus) [inline index.js]: Why is this needed?

**Comment by Avnerus** [comment]
Thank you for this work. Looks great!
I left some comments and questions in the review.

> We had an issue where way too many tiles are loaded in the distance than are needed for this application. There are not visible as they are mostly occluded by buildings in the scene given the camera placement

I think the best bet here would be to implement occlusion culling as in the [threedtiles project](https://github.com/ebeaufay/threedtiles/tree/master) ([service](https://github.com/ebeaufay/threedtiles/blob/1183284d03f7d5211d493e6c31688b67556caf1c/src/tileset/OcclusionCullingService.js), [usage](https://github.com/ebeaufay/threedtiles/blob/1183284d03f7d5211d493e6c31688b67556caf1c/src/tileset/OGC3DTile.js#L530)).

This requires some modification to `three-loader-3dtiles` and also to `loaders.gl`, to allow a callback for an occlusion service. I might be able to get around to this in the near future, but if you need this now you are welcome to take on this task!

File: examples/google-tiles/index.html (hunk 1 above)
Comment (kfarr) [reply examples/google-tiles/index.html:41]: > but there is no option to insert them via the query string.

Correct, the expected behavior in this example is only to place tiles with a long / lat / height specified in the index.html centered at scene origin 0 0 0. Supporting a dynamic, user-specified querystring with lat / long / height was not intended to be part of this example.

File: index.js (hunk 2 above)
Comment (kfarr) [reply index.js]: I'll check with Alex, but I'm pretty sure this is not needed and was only included to improve generation of screenshots for our use case. We can remove this and add elsewhere in our application instead.

File: examples/google-tiles/index.html (hunk 1 above)
Comment (Avnerus) [reply examples/google-tiles/index.html:41]: Got it.
It's maybe OK since the example is anyway not exposed in the demo showcase, but if it works well we should probably consider adding it in the future and also make the example usable without editing the code. Maybe after we have occlusion culling and teleport controls?

Comment (kfarr) [reply examples/google-tiles/index.html:41]: Yes agreed. I've added these tickets to track:
* https://github.com/nytimes/aframe-loader-3dtiles-component/issues/44
* https://github.com/nytimes/aframe-loader-3dtiles-component/issues/43
* https://github.com/nytimes/aframe-loader-3dtiles-component/issues/42

File: index.js (hunk 2 above)
Comment (kfarr) [reply index.js]: Ok this has been removed. Just to make sure I tested with and without and didn't notice any effect.

**Comment by Avnerus** [comment]
Thank you!
